src/main/java/com/exactpro/th2/conn/ConnectivityMessage.java

```diff
@@ -100,6 +112,19 @@ public String toString() {
                 .toString();
     }
 
+    private static <T> void fillParentEventId(IMetadata metaData, Function<EventID.Builder, T> setParentEventId) {
+        String parentId = getParentEventID(metaData);
+        if (parentId != null) {
+            setParentEventId.apply(EventID.newBuilder().setId(parentId));
+        }
+    }
+
+    private static @Nullable String getParentEventID(IMetadata metadata) {
+        return contains(metadata, MetadataProperty.PARENT_EVENT_ID)
+                ? SailfishMetadataExtensions.getParentEventID(metadata).toString()
```

Comment:
If I am correct the `SailfishMetadataExtensions.getParentEventID(metadata)` returns EventID. Why do we need to use `toString()`?
Maybe we will return either EventID or `null`. The EventID is immutable so we don't need to create another one on line 118 in `fillParentEventId` method